### ISSUE TITLE
unable to Unsubscribe  (ActiveMQ) 5.1xx

### ISSUE THREAD

**Issue by warfares** [go-stomp/stomp#109]
Hi 

in case helps, today I tested a very basic sample.

using ActiveMQ 5.14.3. (also AmazonMQ 5.15...)
go version go1.16.3 darwin/amd64 

```go
       conn, err := stomp.Dial("tcp", host, stomp.ConnOpt.Login(user, pwd),
           stomp.ConnOpt.HeartBeatGracePeriodMultiplier(2))
      // handle errors 

       sub, err := conn.Subscribe("/queue/somequeue", stomp.AckClientIndividual)
       // handle errors 

     
      // process just 20 msg.
      for index := 0; index < 20; index++ {
		m := <-sub.C
		conn.Ack(m)
                ... do something with the msg.
	}
	sub.Unsubscribe()
	log.Info("All done")
```

however when get to Unsubscribe it **get lock forever** in this instruction.

subscription.go
```go
	s.closeMutex.Lock()
	for atomic.LoadInt32(&s.state) != subStateClosed {
		s.closeCond.Wait()
	}
	s.closeMutex.Unlock()
```

not sure if Im doing something wrong, or is something with the broker, or the library.
any cases I reported here.

All the Best,
Rodolfo.

**Comment by worg** [comment]
Let me take a look

**Comment by worg** [comment]
I've been a bit busy with my job but hopefully this weekend I'll be able to do some testings

**Comment by warfares** [comment]
Thanks you very much, yep I'll try focus and debug with more detail next week as well ..., in case I could get something..

**Comment by worg** [comment]
Coudn't replicate using ActiveMQ 5.14.13 from this dockerfile
- https://github.com/rmohr/docker-activemq/tree/master/5.14.3-alpine
and this: 

```go
package main

import (
	"log"
	"time"

	stomp "github.com/go-stomp/stomp/v3"
)

var options []func(*stomp.Conn) error = []func(*stomp.Conn) error{
	stomp.ConnOpt.Login("admin", "admin"),
	stomp.ConnOpt.HeartBeatGracePeriodMultiplier(2),
	stomp.ConnOpt.AcceptVersion("1.1"),
	stomp.ConnOpt.UseStomp,
}

const host = `127.0.0.1:61613`

func sendToQueue() {
	conn, err := stomp.Dial("tcp", host, options...)
	if err != nil {
		panic(err)
	}

	// process just 20 msg.
	for index := 0; index < 20; index++ {
		err := conn.Send("/queue/somequeue", "application/json", nil)
		if err != nil {
			log.Println("ERR: ", err.Error())
		}

		time.Sleep(1000)
	}

}

func main() {
	conn, err := stomp.Dial("tcp", host, options...)
	if err != nil {
		panic(err)
	}

	go sendToQueue()

	// handle errors

	sub, err := conn.Subscribe("/queue/somequeue", stomp.AckClientIndividual)
	if err != nil {
		panic(err)
	}
	// handle errors

	// process just 20 msg.
	for index := 0; index < 20; index++ {
		m := <-sub.C
		conn.Ack(m)

		log.Printf("MSG: %+v", m.Body)
	}

	sub.Unsubscribe()
	log.Println("All done")

}
```

**Comment by warfares** [comment]
Thanks for taking the time..., really appreciated.
 
I tried your code and works immediately...., however If I 
1. clean the queue 
2. change the sender for example 40
3. and the consumer just 20 ... 

.... I get the lock.... :-S

I'd expected break the loop and keep  rest 20 in the queue.


when I detect was because I have a queue with 600k msg or more and my consumer(s) process a max of 25k to finish and release resources.


** any-cases Ill continue trying , debuging more carefully in next days.... (in case I'm doing something wrong or a mis understanding from my end.)...  I'd like to use your pkg in large project (and so far is the most clear one for go).


Thanks again.
Rodolfo.

**Comment by worg** [comment]
After some packet capturing I found out that ActiveMQ isn't always sending unsubscribe RECEIPT frames, that's what causes the deadlock (and it's inconsistent, I get it sometimes and sometimes don't)
I think it's probably a server bug, I'll try to repro using another client

**Comment by warfares** [comment]
yep you are absolutely correct, is related to the broker, interesting.... 
(I updated to latest version support by Amazon 5.16.2) and now I observe the inconsistent  ...

** I think issue it could be close now ..., not an problem related directly to the library 
(unless you'd like to handle that cases, this understanding that ActiveMQ is kind of popular broker.. -> a timeout or something in that direction) 

**  in my particular  case, I could sorted out the eventuality.... (exit/kill the process).

Thanks you very much for all the help and great work.
Rodolfo.
